dags/sql/create-function-generic_find_gaps.sql

```diff
@@ -0,0 +1,109 @@
+--noqa: disable=TMP, PRS
+CREATE OR REPLACE FUNCTION public.generic_find_gaps(
+    start_date date,
+    end_date date,
+    id_col text,
+    dt_col text,
+    sch_name text,
+    tbl_name text,
+    gap_threshold interval,
+    default_bin interval,
+    id_col_dtype anyelement DEFAULT NULL::int
+)
+RETURNS TABLE (
+    sensor_id_col anyelement, gap_start timestamp, gap_end timestamp
+)
+LANGUAGE plpgsql
+COST 100
+VOLATILE
+
+AS $BODY$
+DECLARE
+   resulttype regtype := pg_typeof(id_col_dtype);
+BEGIN
+    RETURN QUERY EXECUTE FORMAT($$
+        WITH raw AS (
+            SELECT DISTINCT
```

Comment:
I guess one "caveat" for this type of query is if the sensor has a gap larger than [start_date, end_date), then it won't show up at all eh?